Lib/test/test_import/__init__.py

```diff
@@ -804,104 +804,133 @@ def test_issue105979(self):
                       str(cm.exception))
 
     def test_script_shadowing_stdlib(self):
-        with os_helper.temp_dir() as tmp:
-            with open(os.path.join(tmp, "fractions.py"), "w", encoding='utf-8') as f:
-                f.write("import fractions\nfractions.Fraction")
-
-            expected_error = (
-                rb"AttributeError: module 'fractions' has no attribute 'Fraction' "
-                rb"\(consider renaming '.*fractions.py' since it has the "
-                rb"same name as the standard library module named 'fractions' "
-                rb"and the import system gives it precedence\)"
+        script_errors = [
+            (
+                "import fractions\nfractions.Fraction",
+                rb"AttributeError: module 'fractions' has no attribute 'Fraction'"
+            ),
+            (
+                "from fractions import Fraction",
+                rb"ImportError: cannot import name 'Fraction' from 'fractions'"
             )
+        ]
+        for script, error in script_errors:
+            with os_helper.temp_dir() as tmp:
+                with open(os.path.join(tmp, "fractions.py"), "w", encoding='utf-8') as f:
+                    f.write(script)
+
+                expected_error = error + (
+                    rb" \(consider renaming '.*fractions.py' since it has the "
+                    rb"same name as the standard library module named 'fractions' "
+                    rb"and the import system gives it precedence\)"
```

Comment:
```suggestion
                    rb"and prevents importing that standard library module\)"
```

Using `it` a second time is ambiguous here, since unlike the first use, there are now three preceding noun phrases (".*fractions.py", "the standard library module name 'fractions'", and "the import system"), and the one we actually mean is the one that is farthest away from the pronoun. Technically correct in a parse tree (I think), but tricky for the human brain to follow.

The more direct statement of the consequences not only eliminates the ambiguous pronoun, but also means the reader doesn't need to work out why the import system giving something precedence matters to them.